src/test/java/com/ot/restaurant/repositories/DummyGenericRepositoryImpl.java

```diff
@@ -0,0 +1,3 @@
+package com.ot.restaurant.repositories;
+
+public class DummyGenericRepositoryImpl extends GenericRepositoryImpl {}
```

Comment:
I think this should be:
```
public class DummyGenericRepositoryImpl extends GenericRepositoryImpl <DummyBaseEntity> {}
```

That way you don't need to cast to `(Base)` in other tests